### PR TITLE
ARM: misc enhancements & fixes for LTS (Part 1)

### LINKED ISSUE #36278: ARM: Cortex-M: SysTick priority is not initialized if the SysTick is not used
**Describe the bug**
The SysTick IRQ priority level is never set if the SysTick is not used as the system timer. The IRQ however is always enabled, so if it is accidentally pended, the ISR will run at the highest priority level, where asserts are not handled properly (escalation to hard fault). In detail, instead of getting an expected "unhandled IRQ" error, which we may have routines to handle, we get a hard fault with generic CPU fault, without any additional information.

**Impact**
For the issue to occur, one needs to accidentally set the SysTick IRQ to pending. This is not normally done by software, so in regular applications this is not an alarming bug, at all. But it might be an issue with Safety critical operations, where the fact that the kernel crash appears with the wrong type could impact device recovery.

### LINKED ISSUE #36737: Cortex M23: "swap_helper.S:223: Error: invalid offset, value too big (0x0000009C)"
**Describe the bug**
For a new Cortex M23 soc in development, when enabling CONFIG_DEBUG_THREAD_INFO=y,  the assembler gives me the error:
`/zephyr/arch/arm/core/aarch32/swap_helper.S:223: Error: invalid offset, value too big (0x0000009C)`

The following patch fixes the issue for me:
```
diff --git a/arch/arm/core/aarch32/swap_helper.S b/arch/arm/core/aarch32/swap_helper.S
index 70b009e3442f..460da5285385 100644
--- a/arch/arm/core/aarch32/swap_helper.S
+++ b/arch/arm/core/aarch32/swap_helper.S
@@ -208,7 +208,7 @@ out_fp_endif:
      * (We clear the arch.basepri field after restoring state)
      */
 #if (defined(CONFIG_CPU_CORTEX_M0PLUS) || defined(CONFIG_CPU_CORTEX_M0) || \
-       defined(CONFIG_CPU_CORTEX_M1)) && _thread_offset_to_basepri > 124
+       defined(CONFIG_CPU_CORTEX_M1) || defined(CONFIG_CPU_CORTEX_M23)) && _thread_offset_to_basepri > 124
     /* Doing it this way since the offset to thread->arch.basepri can in
      * some configurations be larger than the maximum of 124 for ldr/str
      * immediate offsets.
```

although I wonder if it would not be appropriate to test  `defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)`

**To Reproduce**
I don't think there are any Cortex M23 soc in the main repo, so I don't have an easy recipe, but I guess any Cortex M23 with CONFIG_DEBUG_THREAD_INFO=y might trigger the issue.

**Expected behavior**
The assembler should run successfully.

**Impact**
Does not compiles without a patch.

**Logs and console output**
```
[build] [1/9  11% :: 0.021] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj
[build] FAILED: zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj 
[build] ccache /home/maxime/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.6.0-566-g98f879bad65b -DKERNEL -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/maxime/projets/zephyr/zephyr/kernel/include -I/home/maxime/projets/zephyr/zephyr/arch/arm/include -I/home/maxime/projets/zephyr/zephyr/include -Izephyr/include/generated -I/home/maxime/projets/hermes/extension/soc/arm/renesas_ra/ra2l1 -I/home/maxime/projets/zephyr/zephyr/drivers -I/home/maxime/projets/hermes/extension/soc/arm/renesas_ra/common -I/home/maxime/projets/zephyr/modules/hal/cmsis/CMSIS/Core/Include -isystem /home/maxime/projets/zephyr/zephyr/lib/libc/minimal/include -isystem /home/maxime/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.2.0/include -isystem /home/maxime/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.2.0/include-fixed -g -Og -imacros /home/maxime/projets/hermes/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m23 -mthumb -mabi=aapcs -xassembler-with-cpp -imacros /home/maxime/projets/zephyr/zephyr/include/toolchain/zephyr_stdint.h -D_ASMLANGUAGE -Wno-unused-but-set-variable -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/maxime/projets/hermes=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/maxime/projets/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/maxime/projets/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -nostdinc -MD -MT zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj -MF zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj.d -o zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj -c /home/maxime/projets/zephyr/zephyr/arch/arm/core/aarch32/swap_helper.S
[build] /home/maxime/projets/zephyr/zephyr/arch/arm/core/aarch32/swap_helper.S: Assembler messages:
[build] /home/maxime/projets/zephyr/zephyr/arch/arm/core/aarch32/swap_helper.S:223: Error: invalid offset, value too big (0x0000009C)
[build] /home/maxime/projets/zephyr/zephyr/arch/arm/core/aarch32/swap_helper.S:225: Error: invalid offset, value too big (0x0000009C)
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.12.4
 - Commit SHA: 98f879bad65b (HEAD -> main) "boards: thingy53: start network core before 802.15.4 driver"

### PR DESCRIPTION
**Pull request by ioannisg**
Collection of fixes and improvements for the ARM architecture port

~~In progress, but folks are welcome to put comments, suggestions.~~

Taking this out of draft. Rename to "part 1". There will be a second part for ARM fixes for the LTS.

Fixes #36278 
Fixes #36737

### CODE REVIEW

**Review by julien-massot — APPROVED** [review]

**Review by bbolen — CHANGES_REQUESTED** [review]

File: arch/arm/core/aarch32/cortex_m/vector_table.S [hunk 1]
```diff
@@ -73,7 +73,7 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 #if defined(CONFIG_MULTITHREADING)
     .word z_arm_pendsv
 #else
-    .word 0
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply arch/arm/core/aarch32/cortex_m/vector_table.S:76]: thanks, fixed.

File: arch/arm/core/aarch32/Kconfig [hunk 2]
```diff
@@ -132,8 +132,8 @@ config RUNTIME_NMI
 config PLATFORM_SPECIFIC_INIT
 	bool "Enable platform (SOC) specific startup hook"
 	help
-	  The platform specific initialization code (z_platform_init) is executed
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply arch/arm/core/aarch32/Kconfig:135]: thanks, fixed

File: arch/arm/core/aarch32/cortex_a_r/reset.S [hunk 3]
```diff
@@ -29,7 +29,7 @@ GDATA(z_arm_fiq_stack)
 GDATA(z_arm_abort_stack)
 GDATA(z_arm_undef_stack)
 #if defined(CONFIG_PLATFORM_SPECIFIC_INIT)
-GTEXT(z_platform_init)
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply arch/arm/core/aarch32/cortex_a_r/reset.S:32]: done

File: soc/arm/nxp_lpc/lpc54xxx/gcc/startup_LPC54114_cm4.S [hunk 4]
```diff
@@ -0,0 +1,100 @@
+/*
```
Comment (cfriedt) [inline soc/arm/nxp_lpc/lpc54xxx/gcc/startup_LPC54114_cm4.S:1]: @ioannisg - this one and the one above do not seem exactly miscellaneous. Should they be included here or possibly in a separate PR?

**Review by bbolen — APPROVED** [review]

**Pull request merged**